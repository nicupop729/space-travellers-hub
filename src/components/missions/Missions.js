/* eslint-disable camelcase */
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions } from '../../redux/missions/missions';
import './missions.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionsReducer);
  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

  return (
    <div className="Missions">
      <table className="Missions__table">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Join</th>
          </tr>
        </thead>
        {
          missions.map(({
            mission_id, mission_name, description, reserved = false,
          }) => (
            <tr key={mission_id}>
              <td className="Missions__name"><h1>{ mission_name }</h1></td>
              <td className="Missions__description"><p>{ description }</p></td>
              <td>{ reserved ? (<span className="badge badge--primary">ACTIVE MEMBER</span>) : (<span className="badge badge--secondary">NOT A MEMBER</span>) }</td>
              <td>{ reserved ? (<button type="button" className="app-btn-danger">Leave Mission</button>) : (<button type="button" className="app-btn-ghost">Join Mission</button>) }</td>
            </tr>
          ))
        }
      </table>
    </div>
  );
};

export default Rockets;
