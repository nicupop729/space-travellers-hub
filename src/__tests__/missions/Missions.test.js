import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Missions from '../../components/missions/Missions';

describe('snapshot testing', () => {
  test('snapshot for Missions component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
