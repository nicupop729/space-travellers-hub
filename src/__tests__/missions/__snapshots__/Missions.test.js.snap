// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Missions component 1`] = `
<div
  className="Missions"
>
  <table
    className="Missions__table"
  >
    <thead>
      <tr>
        <th>
          <h1>
            Mission
          </h1>
        </th>
        <th>
          <h1>
            Description
          </h1>
        </th>
        <th>
          <h1>
            Status
          </h1>
        </th>
        <th>
          <h1>
            Join
          </h1>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
